neuralcompression/models/scale_hyperprior.py

```diff
@@ -394,15 +394,33 @@ def update(self, force=False):
         hyper_bottleneck_updated = self.hyper_bottleneck.update(force=force)
         return image_bottleneck_updated | hyper_bottleneck_updated
 
+    def _assert_on_cpu(self, force_cpu: bool):
```

Comment:
Could we have a function that checks if we’re on CPU that returns true or false and have the exception logic in the main function? That makes this cleaner rather than a branched return.